frontend/src/app/_services/schedule.service.ts

```diff
@@ -20,13 +19,145 @@ export interface ScheduleStatus {
   lastUpdate?: Date;
 }
 
-export interface ScheduleData {
-  program: ProgramItem[],
-  people: ProgramPerson[]
+const tzoffset: number = new Date().getTimezoneOffset();
+
+export class ScheduleEvent {
+  constructor (item: ProgramItem) {
+    this.id = item.id;
+    this.title = item.title;
+    this.description = item.desc;
+    this.tags = item.tags || [];
+    this.start = new Date(item.timestamp);
+    this.mins = parseInt(item.mins, 10);
+    this.location = item.loc;
+    this.people = [];
+    this.tempPeople = item.people;
+  }
+
+  link(peopleMap: {[_: string]: SchedulePerson}): void {
+    if (this.tempPeople) {
+      this.people = this.tempPeople.
+        map(({id, name}) => ({person: peopleMap[id], isModerator: name.endsWith('(moderator)')})).
+        filter(({person}) => person);
+    }
+    this.tempPeople = undefined;
+  }
+
+  getDateString(): string {
+    if (!this.dateString) {
+      this.dateString = this.start.toLocaleDateString(undefined, {'weekday': 'long', month: 'long', day: 'numeric', year: 'numeric'});
+    }
+    return this.dateString;
+  }
+
+  getTimeString(): string {
+    if (!this.timeString) {
+      const localTime = this.start.toLocaleTimeString(undefined, {hour:'numeric', minute: 'numeric', hour12: true});
+      // Check only works when outside daylight savings time. We could
```

Comment:
my 'v2' suggestion is for people to set the schedule offset in a config, and then use a value from there. this takes care of any con that is not over a time change weekend (arisia never is, but clearly our v2 code should be useable by everyone ;) )